packages/server/api/src/app/mcp/mcp-server/mcp-server.ts

```diff
@@ -253,6 +269,135 @@ async function addFlowToServer(
+    const actionPropertiesEntries = Object.entries(actionProperties)
+
+    const parsedParameters = await actionPropertiesEntries.reduce(async (accPromise, [propertyName, propertyDefinition]) => {
+        const acc = await accPromise
+        const needsDynamicResolution = propertyDefinition.type === PropertyType.DYNAMIC || 
+            propertyDefinition.type === PropertyType.DROPDOWN || 
+            propertyDefinition.type === PropertyType.MULTI_SELECT_DROPDOWN
+        
+        let dynamicPropertySchema = 'N/A'
+        
+        if (needsDynamicResolution) {
+            const propertyResolutionResult = await userInteractionWatcher(logger)
+                .submitAndWaitForResponse<EngineHelperResponse<EngineHelperPropResult>>({
+                jobType: UserInteractionJobType.EXECUTE_PROPERTY,
+                projectId,
+                propertyName,
+                actionOrTriggerName: actionMetadata.name,
+                input: acc,
+                piece: piecePackage,
+                sampleData: {},
+            })
+            dynamicPropertySchema = JSON.stringify(propertyResolutionResult.result, null, 2)
+        }
+
+        const parameterExtractionPrompt = `
+You are an expert at understanding API schemas and filling out properties based on user instructions.
+
+TASK: Fill out the property "${propertyName}" based on the user's instructions.
+
+USER INSTRUCTIONS:
+${userInstructions}
+
+PROPERTY DETAILS:
+- ${JSON.stringify(propertyDefinition, null, 2)}
+
+PROPERTY SCHEMA:
+${dynamicPropertySchema}
+
+IMPORTANT:
+- For DYNAMIC properties, for each value, wrap the keys inside the options property inside an object with the same property name, and assign the array to the property name. For example, if the property is "values", return: { "values": [ { ...optionKeys }, ... ] }.
+- For dropdown properties, select values from the provided options array only
+- For ARRAY properties with nested properties (like A, B, C), return: [{"A": "value1", "B": "value2", "C": "value3"}]
+- Return valid JSON for complex types, raw values for simple types
+- Must include all required properties, even if the user does not provide a value
+- For CHECKBOX properties, return true or false
+- For SHORT_TEXT and LONG_TEXT properties, return string values
+- For NUMBER properties, return numeric values
+- For DATE_TIME properties, return date strings in ISO format (YYYY-MM-DDTHH:mm:ss.sssZ)
+- If the property is not required, return SKIP_PROPERTY if the user does not provide a value
+- Use actual values from user instructions when available
+
+CONTEXT:
+- Previously filled properties: ${JSON.stringify(acc, null, 2)}
+
+INSTRUCTIONS:
+1. Extract the property value from user instructions
+2. For required properties without clear instructions, use defaults or make reasonable assumptions
+3. Return raw values for simple types, valid JSON for complex types
+
+RESPONSE FORMAT:
+- Return only the value (raw for simple types, JSON for complex types)
```

Comment:
I don't think you should specific the format and then try to parse it, most of AI models allow you to specify JSON (Zod Schema) and it will answer it based on your schema.